src/node_crypto.cc

```diff
@@ -4722,6 +4767,55 @@ void Verify::VerifyFinal(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(verify_result);
 }
 
+void VerifyOneShot(const FunctionCallbackInfo<Value>& args) {
+  ClearErrorOnReturn clear_error_on_return;
+  Environment* env = Environment::GetCurrent(args);
+
+  unsigned int offset = 0;
+  ManagedEVPPKey key = GetPublicOrPrivateKeyFromJs(args, &offset);
+  if (!key)
+    return;
+
+  switch (EVP_PKEY_id(key.get())) {
+    case EVP_PKEY_ED25519:
+    case EVP_PKEY_ED448:
+      break;
+    default:
+      return env->ThrowError("Unsupported key type");
```

Comment:
Is it possible to do this check at the JavaScript layer? Either way, it would be great to have an `ERR_*` code for this.

Reply:
Probably not reliably and not without crossing the JS<->C++ boundary an extra time. Is there a way to throw errors with codes set from C++?

Reply:
Added, but I noticed that the `name` for error objects produced from C++ are not formatted the same as those generated from JS. JS includes the code in brackets after the error type. C++ does not.

Reply:
I would like an error code, if its not too much trouble. I have a PR almost done that adds a lot more error codes, but it didn't work everywhere, and I had to put it aside while doing other work. Every new throw of a string is something I'll have to convert to setting .code sometime in the future.